### PR TITLE
CLN: Move info

### PR DESCRIPTION
**Pull request by MarcoGorelli**
precursor to xref #31796. Have moved `DataFrame.info` code into `pandas/io/formats/info.py`, along with tests

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

File: pandas/conftest.py [hunk 1]
```diff
@@ -111,6 +111,35 @@ def axis_series(request):
     return request.param
 
 
+@pytest.fixture
+def datetime_frame():
```
Comment (jbrockmendel) [inline pandas/conftest.py]: can we clarify that this is datetime index with float values?

Comment (MarcoGorelli) [reply pandas/conftest.py]: E.g. rename it to `float_frame_dt_index`? That would involve changing a lot of files, but sure, can do

**Comment by simonjayhawkins** [comment]
@MarcoGorelli can you merge master to resolve conflicts

**Comment by MarcoGorelli** [comment]
@simonjayhawkins sure, done

File: pandas/io/formats/info.py [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+from pandas.io.formats import format as fmt
+from pandas.io.formats.printing import pprint_thing
+
+
+def _put_str(s, space):
+    return str(s)[:space].ljust(space)
+
+
+def info(
+    data, verbose=None, buf=None, max_cols=None, memory_usage=None, null_counts=None
```
Comment (jreback) [inline pandas/io/formats/info.py]: some point later type these

File: pandas/tests/frame/indexing/test_indexing.py [hunk 3]
```diff
@@ -256,44 +256,48 @@ def inc(x):
         tm.assert_frame_equal(df, expected)
 
     def test_getitem_boolean(
-        self, float_string_frame, mixed_float_frame, mixed_int_frame, datetime_frame
+        self,
```
Comment (jreback) [inline pandas/tests/frame/indexing/test_indexing.py]: is this orthogonal?

**Review by jreback — CHANGES_REQUESTED** [review]
pls don't mix fixturing with a move. this is too confusing. just do these move in this PR; you can make another one for the other changes.

File: pandas/conftest.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/conftest.py]: Seems Jeff would prefer this to be done separately, so have reverted for now

**Comment by MarcoGorelli** [comment]
@jreback sure, have reverted renaming the fixture and have added type annotations

File: pandas/tests/io/formats/test_info.py [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+     0   a       2 non-null      int64
+    dtypes: int64(1)
+    memory usage: {bytes} bytes
+    """
+    )
+    assert result == expected
+
+
+def test_info_wide():
+    from pandas import set_option, reset_option
```
Comment (jreback) [inline pandas/tests/io/formats/test_info.py]: can you import these at the top

File: pandas/conftest.py [hunk 5]
```diff
@@ -112,6 +112,35 @@ def axis_series(request):
     return request.param
 
 
+@pytest.fixture
```
Comment (jreback) [inline pandas/conftest.py]: i would rather move this into test_info.py as this overlaps with our generalized fixtures for frames (this is different and only used in 1 place).

**Review by jreback — CHANGES_REQUESTED** [review]
looks ok, ping on green.

**Comment by MarcoGorelli** [comment]
@jreback sure, have moved all imports to the top and have reverted moving the fixture

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @MarcoGorelli 

move PRs have to be exactly a move only (and whatever fixes are needed in the target files  to get green CI), otherwise very difficult / impossible to reveiw.

**Pull request merged**